ambry-store/src/main/java/com.github.ambry.store/PersistentIndex.java

```diff
@@ -625,16 +690,19 @@ FindInfo findEntriesSince(FindToken token, long maxTotalSizeOfEntries) throws St
         // check journal
         List<JournalEntry> entries = journal.getEntriesSince(offsetToStart, storeToken.getInclusive());
         logger.trace("Journal based token, Time used to get entries: {}", (time.milliseconds() - startTimeInMs));
+        ConcurrentSkipListMap<Offset, IndexSegment> indexSegments = validIndexSegments;
 
-        if (entries != null) {
+        if (entries != null && journal.getFirstOffset().compareTo(offsetToStart) <= 0) {
           startTimeInMs = time.milliseconds();
           logger.trace(
               "Index : " + dataDir + " retrieving from journal from offset " + offsetToStart + " total entries "
                   + entries.size());
           Offset offsetEnd = offsetToStart;
           long currentTotalSizeOfEntries = 0;
           for (JournalEntry entry : entries) {
-            IndexValue value = findKey(entry.getKey());
+            IndexValue value =
+                findKey(entry.getKey(), new FileSpan(entry.getOffset(), getCurrentEndOffset(indexSegments)),
```

Comment:
That's correct. As suggested, I have called this out via a comment so that it is clear when reading code.